src/Stratis.Bitcoin/Utilities/RetryStrategy.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NLog.LayoutRenderers.Wrappers;
+
+namespace Stratis.Bitcoin.Utilities
+{
+    public static class RetryStrategy
+    {
+        /// <summary>
+        /// Execute logic that should be retried if failure defined by <c>TException</c> occurs.
+        /// </summary>
+        /// <param name="retryOptions">Retry options, including number of retries and delay between them.</param>
+        /// <param name="actionToExecute">Logic to be executed.</param>
+        /// <param name="logger">Optional logger.</param>
+        public static void Run(RetryOptions retryOptions, Action actionToExecute, ILogger logger = null)
+        {
+            if (retryOptions == null || retryOptions.RetryCount == 0)
+            {
+                actionToExecute();
+                return;
+            }
+
+            for (int i = 0; i <= retryOptions.RetryCount; i++)
```

Comment:
Can change this to `for (int i = 0; i <= 1 + (retryOptions?.RetryCount ?? 0); i++)` and remove the preceding code. A `RetryCount` of 0 should mean "execute once" and `RetryCount` of 1 should mean "execute up to twice".